### PR TITLE
Generic serialization of field and group elements (contains #220) (depends on #233)

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

File: libzeth/core/field_element_utils.tcc [hunk 1]
```diff
@@ -19,10 +20,93 @@
 namespace libzeth
 {
 
+namespace impl
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply libzeth/core/field_element_utils.tcc:23]: Yes.  It's a name I've seen in a few projects to separate internal details (in this case the fact that C++ forces use to make a class to make the templating work).  Happy to rename or move everything back to libzeth if you prefer.

[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply libzeth/core/field_element_utils.tcc:23]: Thanks, that's what I thought.
What about renaming this to `internal` like in: https://github.com/clearmatics/libff/issues/21#issue-652221590? I find it more explicit

[the remark this answers is not in the document]
Comment (dtebbs) [reply libzeth/core/field_element_utils.tcc]: renamed

[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply libzeth/core/field_element_utils.tcc]: Thanks

**Comment by AntoineRondelet** [comment]
LGTM, I changed the base branch to point to develop, in which I merged the `fix-type-naming` branch

**Pull request merged**